packages/core/src/telemetry/service.ts

```diff
@@ -1,38 +1,91 @@
 import Conf from "conf";
 import { randomBytes } from "crypto";
+import * as fs from "fs";
 import { createHash } from "node:crypto";
+import os from "os";
+import PQueue from "p-queue";
 import pc from "picocolors";
 
 import { type Options } from "@/config/options";
-import { getAnonymousMeta } from "@/telemetry/anonymous-meta";
+import { postEvent } from "@/telemetry/post-event";
 import { getGitRemoteUrl } from "@/telemetry/remote";
 
 type TelemetryEvent = {
   eventName: string;
   payload: object;
 };
 
+type Context = {
+  sessionId: string;
+  anonymousId: string;
+  projectId: string;
+  meta: AnonymousMeta;
+};
+
+type SerializableTelemetryEvent = TelemetryEvent & Context;
+
 type TelemetryConfig = {
   enabled: boolean;
   notifiedAt: string;
   anonymousId: string;
   salt: string;
 };
 
+type AnonymousMeta = {
+  // Software information
+  systemPlatform: NodeJS.Platform;
+  systemRelease: string;
+  systemArchitecture: string;
+  // Machine information
+  cpuCount: number;
+  cpuModel: string | null;
+  cpuSpeed: number | null;
+  memoryInMb: number;
+  // TODO: detect docker
+  // isDocker: boolean;
+  // isNowDev: boolean;
+  // isWsl: boolean;
+  // isCI: boolean;
+  // ciName: string | null;
+  // nextVersion: string;
+};
+
 export class TelemetryService {
   private readonly conf: Conf<TelemetryConfig>;
   private readonly sessionId: string;
   private readonly TELEMETRY_DISABLED: boolean;
   private rawProjectId: string | null = null;
+  private queue = new PQueue({ concurrency: 1 });
+  private distDir: string;
+  private controller = new AbortController();
+  private events: SerializableTelemetryEvent[] = [];
+  private context?: Context;
+  private metadata?: AnonymousMeta;
 
   constructor({ options }: { options: Options }) {
-    this.conf = new Conf({ projectName: "ponder", cwd: options.ponderDir });
+    this.conf = new Conf({
+      projectName: "ponder",
+      cwd: options.ponderDir,
+      configName: "telemetry-config",
+    });
+    this.distDir = options.ponderDir;
```

Comment:
I would probably create a class property for `this.options: Options` rather than this. Then in the detachedFlush method, can `path.join(this.options.ponderDir, "events.json")` or something similar.